src/targets/Logary.Targets.Stackdriver/Target_Stackdriver.fs

```diff
@@ -64,15 +69,50 @@ module internal Impl =
   | LogLevel.Warn -> LogSeverity.Warning
   | LogLevel.Verbose -> LogSeverity.Debug
 
+  let rec toProtobufValue = function
+    | Value.String s -> WellKnownTypes.Value.ForString(s)
+    | Value.Bool b -> WellKnownTypes.Value.ForBool(b)
+    | Value.Float f -> WellKnownTypes.Value.ForNumber(f)
+    | Value.Int64 i -> WellKnownTypes.Value.ForNumber(float i)
+    | Value.BigInt b -> WellKnownTypes.Value.ForNumber(float b)
+    | Value.Binary (bytes, contentType) -> 
+      // could not find a good way to convert these, will just send null value instead
+      WellKnownTypes.Value.ForNull()
+    | Value.Fraction (numerator,denominator) -> WellKnownTypes.Value.ForNumber(float(numerator/denominator))
+    | Value.Object values -> 
+      values |> Map.toSeq 
+      |> Seq.fold (fun (s : WellKnownTypes.Struct) (k,v) -> s.Fields.[k] <- toProtobufValue v; s) (WellKnownTypes.Struct()) 
+      |> WellKnownTypes.Value.ForStruct
+    | Value.Array values -> 
+      let valueArr = values |> List.map toProtobufValue |> List.toArray
+      WellKnownTypes.Value.ForList valueArr
+
+  and addToStruct (s : WellKnownTypes.Struct) (k : PointName, (Field(value,units))) : WellKnownTypes.Struct = 
+    s.Fields.[string k] <- toProtobufValue value
+    s
+
   let write (m : Message) : LogEntry =
-    let values = 
-      match Json.serialize m with
-      | Object values -> values
-      | otherwise -> failwithf "Expected Message to format to Object .., but was %A" otherwise
-    
-    let overrides = Map.add "name" (String <| string m.name) values
-    let json = Google.Protobuf.WellKnownTypes.Struct.Parser.ParseJson (Json.format <| Object overrides)
-    LogEntry(Severity = toSeverity m.level, JsonPayload = json)
+    // labels are used in stackdriver for classification, so lets put context there.
+    // they expect only simple values really in the labels, so it's ok to just stringify them
+    let labels = m.context |> Map.map (fun k v -> Value.toString v) |> Map.toSeq |> dict
+
+    let message = 
+      // TODO: render the template with field values
```

Comment:
I could send the template still, you're right.  But this `message` property is given special treatment in the UI (as shown in the after-collapsed view above) so I want it to be glance-able.